src/_guides/language/effective-dart/design.md

```diff
@@ -1555,14 +1555,19 @@ bool convertToBool(dynamic arg) {
 ### DO use `Future<void>` as the return type of asynchronous members that do not produce values.
 
 When you have a synchronous function that doesn't return a value, you use `void`
-as the return type. The asynchronous equivalent is `Future<void>`.
+as the return type. The asynchronous equivalent for a method that doesn't
+produce a value, but that the caller might need to await, is `Future<void>`.
 
 You may see code that uses `Future` or `Future<Null>` instead because older
 versions of Dart didn't allow `void` as a type argument. Now that it does, you
 should use it. Doing so more directly matches how you'd type a similar
 synchronous function, and gives you better error-checking for callers and in the
 body of the function.
 
+For functions that are "fire and forget", which neither return a useful
+value nor need to have their asynchronous work finished before the caller
+continues, use a return type of `void`.
```

Comment:
I'd drop the "fire and forget" (I don't think that we are trying to define/re-enforce the definition of such a term), and make it clear that we're talking about `async` functions (even if we are in a section discussing asynchronous members). How about:

> For async functions that do not return a useful value and do not need to have their asynchronous work awaited on, use a return type of `void`.

Reply:
I thought functions marked `async` automatically had Future return values. Plus maybe we don't want to imply that this applies only to those kinds of functions. Instead of "async", should we say "asynchronous"?

Reply:
Oh yes, sorry, I meant "asynchronous".

Reply:
Done